src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/WorkspaceExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default
+{
+    public static class WorkspaceExtensions
+    {
+        /// <summary>
+        /// Tries to find a document given the <see cref="SyntaxTree"/> provided.
+        /// </summary>
+        /// <param name="sln">The solution to search in.</param>
+        /// <param name="tree">The tree to match.</param>
+        /// <param name="document">The resulting document.</param>
+        /// <returns>Whether the search was successful or not.</returns>
+        public static bool TryGetDocument(this Solution sln, SyntaxTree? tree, [MaybeNullWhen(false)] out Document document)
+        {
+            if (tree is null || sln is null)
+            {
+                document = null;
+                return false;
+            }
+
+            foreach (var project in sln.Projects)
+            {
+                var doc = project.GetDocument(tree);
```

Comment:
If I copy and paste the same class file from projectA into projectB then this will find the first instance right?

Reply:
No - it's still a separate document